### PR TITLE
fix: handle Outlook HTML content in draft similarity calculation

### PR DESCRIPTION
**Pull request by elie222**
# User description
Fixes draft matching/deletion failing for Microsoft Outlook users.

**TLDR**: The `calculateSimilarity` function wasn't handling Outlook HTML content, causing 0% match rate for Microsoft users vs 21% for Gmail.

## Problem
- Axiom logs showed **Microsoft: 0% perfect matches** out of 6,190 comparisons
- Gmail: 21% perfect matches out of 1,304 comparisons
- Drafts weren't being deleted because similarity scores were too low (0.08-0.75 instead of 1.0)

## Root Cause
When `bodyContentType === 'html'` (Outlook), the `textPlain` field contains HTML that needs to be converted to plain text. The old `calculateSimilarity` passed this directly to `parseReply`, which doesn't handle HTML tags.

## Changes
- Update `calculateSimilarity` to accept `ParsedMessage` objects (with backwards compatibility for plain strings)
- Use same normalization as `isDraftUnmodified`: `convertEmailHtmlToText()` + `stripQuotedContent()`
- Pass full `ParsedMessage` from `draft-tracking.ts` instead of just `textPlain`
- Add integration tests for Outlook HTML scenarios

## Files Changed
- `apps/web/utils/similarity-score.ts` - Updated function signature and normalization
- `apps/web/utils/reply-tracker/draft-tracking.ts` - Pass full ParsedMessage
- `apps/web/utils/similarity-score.test.ts` - Added integration tests

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
```mermaid
graph LR
trackSentDraftStatus_("trackSentDraftStatus"):::modified
calculateSimilarity_("calculateSimilarity"):::modified
cleanupThreadAIDrafts_("cleanupThreadAIDrafts"):::modified
normalizeStoredContent_("normalizeStoredContent"):::added
normalizeProviderContent_("normalizeProviderContent"):::added
CONVERT_EMAIL_HTML_TO_TEXT_("CONVERT_EMAIL_HTML_TO_TEXT"):::added
STRIP_QUOTED_CONTENT_("STRIP_QUOTED_CONTENT"):::added
STRING_SIMILARITY_("STRING_SIMILARITY"):::modified
trackSentDraftStatus_ -- "Now passes full message object to handle Outlook HTML." --> calculateSimilarity_
cleanupThreadAIDrafts_ -- "Now passes full draftDetails to handle HTML and quoted content." --> calculateSimilarity_
calculateSimilarity_ -- "Normalizes stored drafts via HTML-to-text and strip quoted." --> normalizeStoredContent_
calculateSimilarity_ -- "Normalizes provider messages, handling HTML and stripping quoted." --> normalizeProviderContent_
normalizeStoredContent_ -- "Converts stored draft HTML (<br>) to plain text." --> CONVERT_EMAIL_HTML_TO_TEXT_
normalizeStoredContent_ -- "Removes quoted reply blocks from stored draft content." --> STRIP_QUOTED_CONTENT_
normalizeProviderContent_ -- "Converts Outlook HTML provider bodies to plain text." --> CONVERT_EMAIL_HTML_TO_TEXT_
normalizeProviderContent_ -- "Strips quoted replies from provider message content." --> STRIP_QUOTED_CONTENT_
calculateSimilarity_ -- "Compares normalized texts with string-similarity for score." --> STRING_SIMILARITY_
classDef added stroke:#15AA7A
classDef removed stroke:#CD5270
classDef modified stroke:#EDAC4C
linkStyle default stroke:#CBD5E1,font-size:13px
```

Updates the <code>calculateSimilarity</code> function to properly handle HTML content from Outlook and other email providers, resolving issues where draft matching and deletion failed due to incorrect similarity scores. Modifies <code>draft-tracking.ts</code> to pass full <code>ParsedMessage</code> objects to the updated similarity calculation logic.

<details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>fix-up-imports-for</td><td>November 22, 2025</td></tr>
<tr><td>mojkakec12345@gmail.com</td><td>fix-webhook-email-proc...</td><td>July 04, 2025</td></tr></table></details>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1196?tool=ast>(Baz)</a>.

## Summary by CodeRabbit

* **Tests**
  * Added broader, integration-style tests covering similarity comparisons, Outlook/Gmail HTML handling, quoted content, and edge cases for draft comparisons.

* **Refactor**
  * Improved draft similarity and content normalization so drafts from HTML email clients are compared more accurately, reducing false modified/unmodified detections.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/web/utils/similarity-score.ts [hunk 1]
```diff
@@ -1,33 +1,78 @@
 import * as stringSimilarity from "string-similarity";
-import { parseReply } from "@/utils/mail";
+import { convertEmailHtmlToText, parseReply } from "@/utils/mail";
+import {
+  extractDraftPlainText,
+  stripQuotedContent,
+} from "@/utils/ai/choose-rule/draft-management";
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/similarity-score.ts]: Fixed - removed the unused `extractDraftPlainText` import.

File: apps/web/utils/similarity-score.ts [hunk 2]
```diff
@@ -1,33 +1,78 @@
...
+    // Legacy: plain string - use parseReply for backwards compatibility with Gmail
+    const reply = parseReply(providerMessage);
+    normalized2 = reply.toLowerCase().trim();
+  } else {
+    // ParsedMessage - use proper normalization with bodyContentType
+    const text = providerMessage.textPlain || "";
+    normalized2 = normalizeProviderContent(
+      text,
+      providerMessage.bodyContentType,
+    );
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/similarity-score.ts]: Good catch! Implemented the fallback to `textHtml` when `textPlain` is empty, treating it as HTML content.

File: apps/web/utils/reply-tracker/draft-tracking.ts [hunk 3]
```diff
@@ -168,10 +166,10 @@ export async function cleanupThreadAIDrafts({
 
         if (draftDetails?.textPlain) {
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/reply-tracker/draft-tracking.ts]: Fixed - updated the condition to check `draftDetails?.textPlain || draftDetails?.textHtml` so HTML-only drafts are handled. Combined with the `textHtml` fallback in `calculateSimilarity`, this handles the edge case properly.
